src/main/java/sg/ncl/MainController.java

```diff
@@ -3494,6 +3486,514 @@ public String adminEnergy(Model model,
+        } catch (IOException e) {
+            log.warn("Error getting date created {}", e);
+            projectDetails.setDateCreated("");
+        }
+        projectDetails.setEducation(jsonObject.getBoolean("education"));
+        projectDetails.setServiceTool(jsonObject.getBoolean("serviceTool"));
+        projectDetails.setSupportedBy(jsonObject.getString("supportedBy"));
+        JSONArray usages = jsonObject.getJSONArray("projectUsages");
+        for (int i = 0; i < usages.length(); i++) {
+            JSONObject usage = usages.getJSONObject(i);
+            JSONObject usageId = usage.getJSONObject("id");
+            ProjectUsage projectUsage = new ProjectUsage();
+            projectUsage.setId(usageId.getInt("projectDetailsId"));
+            projectUsage.setMonth(usageId.getString("monthYear"));
+            projectUsage.setUsage(usage.getInt("monthlyUsage"));
+            projectDetails.addProjectUsage(projectUsage);
+        }
+        return projectDetails;
+    }
+
+    private List<ProjectDetails> getProjects() {
+        HttpEntity<String> request = createHttpEntityHeaderOnly();
+        ResponseEntity response = restTemplate.exchange(properties.getMonthly(), HttpMethod.GET, request, String.class);
+        JSONArray jsonArray = new JSONArray(response.getBody().toString());
+
+        List<ProjectDetails> projectsList = new ArrayList<>();
+        for (int i = 0; i < jsonArray.length(); i++) {
+            JSONObject jsonObject = jsonArray.getJSONObject(i);
+            ProjectDetails projectDetails = getProjectDetails(jsonObject);
+            projectsList.add(projectDetails);
+        }
+        return projectsList;
+    }
+
+    @GetMapping("/admin/monthly")
+    public String adminMonthly(HttpSession session, Model model) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        List<ProjectDetails> projectsList = getProjects();
+        model.addAttribute("projectsList", projectsList);
+
+        return "admin_monthly";
+    }
+
+    @GetMapping(value = {"/admin/monthly/contribute", "/admin/monthly/contribute/{id}"})
+    public String adminMonthlyContribute(@PathVariable Optional<String> id, HttpSession session, Model model) {
+        if (!validateIfAdmin(session)) {
+            return NO_PERMISSION_PAGE;
+        }
+
+        if (id.isPresent()) {
+            HttpEntity<String> request = createHttpEntityHeaderOnly();
+            ResponseEntity response = restTemplate.exchange(properties.getMonthly() + "/" + id.get(), HttpMethod.GET, request, String.class);
+            JSONObject jsonObject = new JSONObject(response.getBody().toString());
+            ProjectDetails projectDetails = getProjectDetails(jsonObject);
+            model.addAttribute("project", projectDetails);
+        } else {
+            model.addAttribute("project", new ProjectDetails());
+        }
+
+        return "admin_monthly_contribute";
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Define a constant instead of duplicating this literal "admin_monthly_contribute" 3 times. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)